app/controllers/directions_controller.rb

```diff
@@ -0,0 +1,51 @@
+class DirectionsController < ApplicationController
+  def show
+    if params[:user_address]
+      save_restaurant_visit(params[:user_id], params[:restaurant_id], params[:user_address])
+      directions = create_directions_url(params[:user_address], params[:restaurant_address])
+      redirect_to directions
+    elsif params[:user_lat_long]
+      save_restaurant_visit(params[:user_id], params[:restaurant_id], params[:user_lat_long])
+      directions = create_directions_url(params[:user_lat_long], params[:restaurant_address])
+      redirect_to directions
+    end
+  end
+
+  private
+
+  def save_restaurant_visit(user_id, restaurant_id, location)
+    user = User.find(user_id)
+    restaurant = Restaurant.find(restaurant_id)
+    if location.include?('|')
+      latitude = CGI::escape(origin.split('|')[0])
+      longitude = CGI::escape(origin.split('|')[1])
+    else
+      response = service.geocode(location)
+      latitude = response[:results][0][:geometry][:location][:lat]
+      longitude = response[:results][0][:geometry][:location][:lng]
+    end
+    Visit.create(user: user,
```

Comment:
We will want to change this to a Visit.save later when we're doing sad path testing